SupportScripts/build_tpls.py

```diff
@@ -310,25 +311,62 @@ def build_install_nanoflann(
     target = 'nanoflannConfig.cmake'
     return get_full_path_to_cmake_config_dir(install_path, target)
 
+def build_install_stanmath(
+        workdir: str,
+        force_rebuild: bool
+):
+    myname = "stanmath"
+    parent_path = os.path.join(workdir, myname)
+
+    zip_name = "v2.18.0.zip"
+    url = f'https://github.com/stan-dev/math/archive/refs/tags/{zip_name}'
+    zip_path = os.path.join(parent_path, zip_name)
+    unpack_path = os.path.join(parent_path,  "math-2.18.0")
+
+    # prepare
+    remove_everything_if_needed_from(parent_path, force_rebuild)
+
+    if os.path.exists(parent_path):
+        print(f'skipping {parent_path}, because found already. Use -f to rebuid.')
+
+    print("-"*50)
+    print(f'WORKING ON: {myname}')
+    print("-"*50)
+
+    os.mkdir(parent_path)
+
+    # fetch
+    print("1. fetching")
+    urllib.request.urlretrieve(url, zip_path)
 
-def check_compilers():
-    cxx_alias_found = environ.get('CXX')
-    if cxx_alias_found is not None:
-        print(f'found {cxx_alias_found}')
+    # unpacking
+    print("2. unpacking")
+    shutil.unpack_archive(zip_path, parent_path)
+
+    print(f'success getting: {myname}\n')
+
+    print(unpack_path)
+    return os.path.abspath(unpack_path)
+
+
+def check_compilers(compiler_type: str):
+    compiler_alias_found = environ.get(compiler_type)
+    if compiler_alias_found is not None:
+        print(f'found {compiler_alias_found}')
         return
 
-    if not cxx_alias_found:
+    if not compiler_alias_found:
         print(f'-'*50)
         print(f'*** FATAL ERROR ***')
         print(f'-'*50)
-        print(f'CXX not found in environment, I cannot proceed!')
-        print(f'please set CXX to a valid C++ compiler')
+        print(f'{compiler_type} not found in environment, I cannot proceed!')
+        print(f'please set {compiler_type} to a valid C++ compiler')
```

Comment:
this is not right, because it is C++ if compiler_type == CXX else it is C compiler